glibc: update to 2.39.

oreo639 opened the pull request:
<!-- Uncomment relevant sections and delete options which are not applicable -->

#### Testing the changes
- I tested the changes in this PR: **briefly**

[ci skip]

Depends on: https://github.com/void-linux/void-packages/pull/49146
See: https://github.com/void-linux/void-packages/issues/49162

No major packaging changes indicated by upstream:
https://sourceware.org/glibc/wiki/Release/2.39
https://lists.gnu.org/archive/html/info-gnu/2024-01/msg00017.html

<!--
#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**|**NO**
-->

<!-- Note: If the build is likely to take more than 2 hours, please add ci skip tag as described in
https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#continuous-integration
and test at least one native build and, if supported, at least one cross build.
Ignore this section if this PR is not skipping CI.
-->
<!--
#### Local build testing
- I built this PR locally for my native architecture, (ARCH-LIBC)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - aarch64-musl
  - armv7l
  - armv6l-musl
-->

ahesford commented:
I merged #49146 and then did some tests to confirm that glibc would actually build on musl masterdirs without the `localedef` dependency. In 4ed2d581049 I added a guard that will allow glibc to be built on musl without the `C.UTF-8` locale. Please improve on that logic here if you think there is a better option.

oreo639 commented:
> I added a guard that will allow glibc to be built on musl without the C.UTF-8 locale.

Thanks, I added an extra condition in this PR so that if cross compiling to/from the same architecture, it will do the `make -C localedata ...` (since the newly built binary will be run with the newly built glibc).

ahesford commented on srcpkgs/glibc/template:
```diff
@@ -133,7 +128,7 @@ do_install() {
 
 	# The C.UTF-8 locale should always be available
-	if [ -z "${CROSS_BUILD}" ]; then
+	if [ -z "${CROSS_BUILD}" ] || [ "${XBPS_MACHINE%-*}" = "${XBPS_TARGET_MACHINE%-*}" ]; then
```
You could just drop the `[ -z "${CROSS_BUILD}" ]` test here.

oreo639 replied:
Alright.

oreo639 commented:
Merged d52e080941839e39c5425782af4f4b02f141a586
